crates/uv-requirements/src/pyproject.rs

```diff
@@ -554,16 +556,21 @@ pub(crate) fn lower_requirement(
 
 /// Convert a path string to a path section.
 fn path_source(
-    path: String,
+    path: impl AsRef<Path>,
     project_dir: &Path,
     editable: bool,
 ) -> Result<RequirementSource, LoweringError> {
-    let url = VerbatimUrl::parse_path(&path, project_dir)?.with_given(path.clone());
-    let path_buf = PathBuf::from(&path);
+    let url = VerbatimUrl::parse_path(path.as_ref(), project_dir)?
+        .with_given(path.as_ref().to_string_lossy().to_string());
+    let path_buf = path.as_ref().to_path_buf();
     let path_buf = path_buf
         .absolutize_from(project_dir)
-        .map_err(|err| LoweringError::AbsolutizeError(path, err))?
+        .map_err(|err| LoweringError::AbsolutizeError(path.as_ref().to_path_buf(), err))?
         .to_path_buf();
+    if !editable {
+        // TODO(konsti): Support this.
+        return Err(LoweringError::NonEditableWorkspaceDep);
+    }
```

Comment:
When does this occur? I'm looking at this from the perspective of: what regressions could we ship that would accidentally impact users?

Reply:
This branch is only reachable when using `uv.tool.sources` (otherwise source is `None`), so it preview gated

Reply:
Can you please expand the TODO?